### ISSUE TITLE
(Suggestion) Changes to CQB Bionic.

### ISSUE THREAD

**Issue by Noctifer-de-Mortem** [CleverRaven/Cataclysm-DDA#18470]
I think that the skill set of 5 the bionic imposes should stop once the character has passed level 5. If you are level 8 in melee do you forget your experience when you use the Bionic?

Also adding ninjitsu to it would make sense, when you need to silently shank someone with your monomolecular blade.

**Comment by aeoo** [comment]
I agree.  I think the CQB bionic should probably raise the skill level by 1 or 2 points relativistically instead of jumping to an absolute skill level 5.  Or whatever number of points people think is balanced.

It's also possible to make the bionic give diminishing returns.  Maybe if your skill starts out at 0 or 1, it gives you +4, but if your melee skill is already at 8 naturally maybe it only raises your skill level by +1 then.  That way the CQB bionic makes a huge difference on a starting character, but is less important later on, while still providing a tangible benefit in every case.  It's a thought.

I do agree that it's weird how it just locks the skills to 5.

I guess part of the problem is we (or I) don't know how the bionic is supposed to work.  Is it similar to how Neo learns Kung-Fu in the Matrix?  What does it actually do?  I mean, what's the theory/story behind the CQB bionic?  Most bionics are kind of obvious, but something that improves a complex skill is far from obvious at least to me.  I understand how the toolset bionic works.  I understand the lasers in the fingers.  But a machine or a piece of software that improves a complex skill is not something I readily understand.

Plus, if it's truly software for the brain, shouldn't it take 0 body slot space?  If the CQB is software for the brain, it should be easy to install without a knife and the first aid kit, and it should take zero space in the head.

On the other hand, if the CQB bionic is not software for the brain, then what is it?  If we can get some story for it, we'll know how it should work. :)

**Comment by Xnfinity** [comment]
It is kinda odd that you do not gain any skill exp when you have a very low chance to gain martial arts

Though i do like the design of this bionic' it can help so much early game but it will hurt your growth using it and to a seasoned warrior' it is nearly useless

**Comment by illi-kun** [comment]
Re: how this cbm works
I think this metal box installed in your body works as a puppet master, it pushs and pulls your limbs to do right moves in right time, so it coordinates your actions without your participation in this process (and poor kung-fu masters without this magic box have to use their own brain to do the same).

**Comment by aeoo** [comment]
@illi-kun Then how it's presently coded makes perfect sense.  I mean, I understand for gameplay reasons people may not want it to be stuck at 5, but if it's a puppet master box there is no way to combine your own skill with the box.  So it's either the box or you, but not both together.

All this would make a lot more sense if bionics were easier to remove.  Right now bionics are such a pain (sorry for the pun) to remove.  I guess the CQB bionic should only ever be used by someone who does *not* plan to train their unarmed/melee skill?  Or.  Once your skill passes 5 you have to think how to get rid of this bionic, if you started with it.  Good luck trying to remove it, lol.  (80 damage to every body part is brutal, and you better not take any minus to HP traits either, or you're instantly dead from one botched removal procedure; I used to love taking -25% to all HP trait for 4 points, but then I realized it means insta-death from one botched bionic removal, so I never take it now)

I think the CQB bionic as a puppet master box is more of a liability than an asset.  If I were a bionic assassin I wouldn't mess around with the CQB bionic myself because I'd be concerned about the long term impact.  I'm fine with the CQB being a puppet master box in-game, but if that's the case, can we consider removing it as a default bionic from the assassin background?  Then if someone wants to install it, let them.  Although I don't know who'd want it, because almost every character will pass the skill level 5 without much difficulty in the not too distant future.

Also if this bionic is mostly useful for starting characters it should be easier to find, probably.  Otherwise if it's hard to find, then no one will install it by the time they find it.  Why would anyone willingly downgrade their character with a bionic?

**Comment by Noctifer-de-Mortem** [comment]
I don't think ti should just raise your skill level. How is the bionic suppose to know the difference from level 4 to level 5 melee? The raise of stats is fine just don't make it a cap, once a skill is level 6 it is still level 6 when you activate the bionic. The bionic is like a brain transplant that actual as extra neural networks and when activated it give your brain access to level 5 bionics but this does not mean you forget your other abilities. The bionic is a puppet master as in telling your brain how to execute moves ( level 5 moves ) but it does not decide what moves you use, and if you know better moves (level 6 +) you would use them as well.

**Comment by aeoo** [comment]
>I don't think ti should just raise your skill level. How is the bionic suppose to know the difference from level 4 to level 5 melee? The raise of stats is fine just don't make it a cap, once a skill is level 6 it is still level 6 when you activate the bionic.

If the CQB bionic is really a puppet master as illi-kun is saying, then it's a package deal.  You get all of its parameters or none.  I don't think it makes sense to get the CQB's special moves but not its skill stats.

@Noctifer-de-Mortem can you come up with a theory of how a bionic would work such that it gives you its complex martial arts maneuvers, but still allows your own skills to contribute to its performance?

I agree for gameplay reasons that the change you propose is a good one, but how can it be explained to the player?  Cause bionics are not supposed to be magic.  Bionics have to be rational.

**Comment by Noctifer-de-Mortem** [comment]
Think of the bionic as an extra "brain part" with the info about the combat skills. When it is on it is powered with electricity and becomes part of the brain allowing you to "know" the move set, the move set is executed on its own on conscious request (the puppet master part)  making the "knowledge" "subconscious". When the bionic is off, it is as if you had removed a part of the brain that contained the move-set knowledge.

**Comment by aeoo** [comment]
@Noctifer-de-Mortem sounds as plausible to me as the puppet master idea. :)  Although what you describe seems to have a more complex interface because it alters the contents of the subconscious mind.

**Comment by Noctifer-de-Mortem** [comment]
Not much that it alters but behaves as such. The user has no control on the execution of the moves, they can just choose to do the moves and the bionic does it for them.

**Comment by Noctifer-de-Mortem** [comment]
It appears that the offender is:
`if (has_active_bionic("bio_cqb")) {
        unarmed_skill = 5;
        bashing_skill = 5;
        cutting_skill = 5;
        stabbing_skill = 5;
        melee_skill = 5;
    }`
In melee.cpp

Maybe something like this would work? (pesudo code based on limited java experience.)
`if (has_active_bionic("bio_cqb")) {
       if skillLevel("unarmed")<=5;
          { unarmed_skill = 5;}
       if skillLevel("bashing")<=5;
        {bashing_skill = 5;}
       if skillLevel("cutting")<=5;
        {cutting_skill = 5;}
       if skillLevel("stabbing")<=5;
        {stabbing_skill = 5;}
       if skillLevel("melee")<=5;
        {melee_skill = 5;}
    }`

**Comment by WizardOfOoo** [comment]
If it helps, I read the code and clarified the description of the bionic.

https://github.com/CleverRaven/Cataclysm-DDA/pull/20200
